src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/deciders/collator/ActionGrouper.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.deciders.collator;
+
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.actions.Action;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.store.rca.cluster.NodeKey;
+import java.util.List;
+import java.util.Map;
+import org.checkerframework.checker.nullness.qual.NonNull;
+
+/**
+ * A grouping interface to provide different ways to group actions as needed by the {@link
+ * Collator}
+ */
+public interface ActionGrouper {
+
+  /**
+   * Groups the given list of actions by the nodes they impact..
+   *
+   * @param actions The list of actions that need to be grouped.
+   * @return A map of actions grouped by nodes they impact.
+   */
+  @NonNull Map<NodeKey, List<Action>> groupByNodeId(@NonNull final List<Action> actions);
```

Comment:
Can we rename the method name to include `InstanceName` ?

Reply:
yup, renamed it to `groupByInstanceId`.